Tests/MongoSwiftTests/EventLoopBoundMongoClientTests.swift

```diff
@@ -0,0 +1,140 @@
+@testable import MongoSwift
+import Nimble
+import NIO
+import TestsCommon
+
+final class EventLoopBoundMongoClientTests: MongoSwiftTestCase {
+    func testEventLoopBoundDb() throws {
+        let elg = MultiThreadedEventLoopGroup(numberOfThreads: 4)
+        let expectedEventLoop = elg.next()
+
+        try self.withTestClient { client in
+            let elBoundClient = EventLoopBoundMongoClient(client: client, eventLoop: expectedEventLoop)
+            expect(elBoundClient.db(Self.testDatabase).eventLoop) === expectedEventLoop
```

Comment:
can we add a few tests for `MongoDatabase` operations here? As part of that, one that ensures the `MongoCollection` returned from `createCollection` has the right event loop.